### ISSUE TITLE
Bump up deps and versions

### ISSUE THREAD

**Issue by antgamdia** [vmware-tanzu/kubeapps#4661]
I've noticed we have some stuck dependabot's PRs requiring changes, also GKE has updated their _stable_ k8s default version and we haven't updated the CI versions var yet, etc.
Specifically, we should follow the release process with regard to the upgrading process, namely:

- Pending PRs:
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4642
        - Trivial rename in a type name, no user-faced effect whatsoever.
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4656
        -  Add some pkgs in the `transformIgnorePatterns` when running jest.
    - [ ] https://github.com/vmware-tanzu/kubeapps/pull/4605
        - Use a new options object and fix tests. 
- Frontend-related
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4664 
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4665
- Project-related
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4666
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4670
- CI-related
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4667
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4668
- Backend-related
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4669
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4671
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4674
    - [x] https://github.com/vmware-tanzu/kubeapps/pull/4687

**Comment by absoludity** [comment]
I had added some to the Kubeapps project - (for eg, if you look at the first bump PR #4642 above, you'll see it has the vmware-tanzu/kubeapps project. I had expected that would make it visible on the project page, but looks like that currently only shows issues. Worth checking as it would be nice to add such bump PRs requiring work to the board without needing a separate issue to track them (haven't checked, but will).

> Specifically, we should follow the release process with regard to the upgrading process

I did specifically mention in the standup before I did the last release that I'd skip the upgrading process to just do a quick release, knowing that we'd need to do them afterwards. You don't need to pick them up, but feel free :)

**Comment by antgamdia** [comment]
Yep, I noticed, but during the iteration planning we assumed it was an error and we added the filer "is:issue". Nevertheless, I wanted to have a quick summary of the required updates at a glance in this case, but, for the future, +1 to use PR-derived cards in the project so that we can track the progress for non-trivial stuff.

> I did specifically mention in the standup before I did the last release that I'd skip the upgrading process to just do a quick release, knowing that we'd need to do them afterwards. You don't need to pick them up, but feel free :)

Sure! That's what we decided given that we needed to release that version asap. However, it is something I wanted to do as a Monday-beginning-of-iteration task while catching up with everything :P
